src/decide.ts

```diff
@@ -225,20 +226,9 @@ async function assessCandidateHelper(
 	if (hashesToExclude.includes(candidateMeta.infoHash)) {
 		return { decision: Decision.INFO_HASH_ALREADY_EXISTS };
 	}
-
-	const partialSizeMatch = compareFileTreesPartialIgnoringNames(
-		candidateMeta,
-		searchee,
-	);
-	if (!partialSizeMatch && matchMode === MatchMode.PARTIAL) {
-		return { decision: Decision.SIZE_MISMATCH };
-	}
 	const sizeMatch = compareFileTreesIgnoringNames(candidateMeta, searchee);
-	if (!sizeMatch && matchMode !== MatchMode.PARTIAL) {
-		return { decision: Decision.SIZE_MISMATCH };
-	}
-
```

Comment:
why did we need to change the matching logic?

Reply:
It seemed more appropriate to check for other matching before checking for partial, and it still makes sense to me. On a prior iteration where I was trying to revert the entire v6 behavior for qbittorrent, it also kept that branch of logic isolated at the end, preferring the other match types based on the order of "best-match-scenario" (and matches that qbittorrent supported would return before doing anything that it wasn't supporting) without the necessity of extra CPU cycles.

@ShanaryS helped with the partial flow that was moved to the end, so if there are questions concerning the more specific ordering in MATCH_PARTIAL returns, they'd be able to shed some light.

Reply:
I remember this being in the original draft of the partial matching PR. Seems fine to me

Reply:
i'm curious if checking something like the piece size or some other meta data to determine rechecking would be a way for us to determine if rechecking risky was necessary.

Reply:
The most complete solution would probably be to recheck non perfectMatches and unpause if at 100% when done.

Reply:
I think the most easy way to address this is to offer two PARTIAL decisions, SAFE_PARTIAL where linking is disabled or not possible for that data - and a perfect/direct match to the original data can be made, and normal PARTIAL.

but then we would probably have to introduce that as a `matchMode` as well.

You did say that that setup works if linking fails with partial, right? I'm not misunderstanding?